src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -273,4 +370,77 @@ async Task LogDiagnosticsAsync(ImmutableArray<DiagnosticLogItem> diagnosticLogIt
             await toastErrorReporter.ReportErrorAsync(worstLspMessageKind, message, cancellationToken);
         }
     }
+
+    protected async ValueTask BeginLoadingProjectAsync(string projectPath, ProjectId primordialProjectId)
+    {
+        using (await _gate.DisposableWaitAsync(CancellationToken.None))
+        {
+            // Note that we don't expect consecutive calls to begin loading the same project.
+            // If that starts happening (i.e. Add throws here), we need to revisit that assumption.
```

Comment:
It turns out it is pretty simple to make this happen in the 'LoadProjects' overloads below, by simply running the 'Open Solution' command multiple times.

Maybe there should be a behavior here where selecting the same solution as before is a no-op, and selecting a different solution causes projects which are not part of the new solution to be unloaded.

In general, perhaps the most reasonable behavior here is that if the key is already present, we are already working on loading this. So just return without doing anything. Probably do not even queue a reload.